### PR TITLE
New error series in Overview graphs

### PR DESCRIPTION
**Pull request by leandroberetta**
This PR is a work in progress, it adds:

-  Y axis enabled allowing comparisons between namespaces rps and errors. It is easier now how to tell which namespaces have more traffic
- New time series indicating errors (5xx and 4xx)

overview

Small detail, I always observed that the cards stack over each other vertically but horizontally there is a 10px space between them, I've just tried to give the 10px vertically for more consistency, opinions?.

### CODE REVIEW

**Comment by lucasponce** [comment]
> Small detail, I always observed that the cards stack over each other vertically but horizontally there is a 10px space between them, I've just tried to give the 10px vertically for more consistency, opinions?.

+1, thanks for addressing this style issue, does it work in all cases ? I remember some edge cases that had some weird spacing.

But, let me know when you'd like we start reviewing the PR and I can test it locally.

**Comment by leandroberetta** [comment]
> > Small detail, I always observed that the cards stack over each other vertically but horizontally there is a 10px space between them, I've just tried to give the 10px vertically for more consistency, opinions?.
> 
> +1, thanks for addressing this style issue, does it work in all cases ? I remember some edge cases that had some weird spacing.

I tested in different situations and seems to work well!.

> 
> But, let me know when you'd like we start reviewing the PR and I can test it locally.

@lucasponce it's out of "do not merge" now, you can review it now.

File: src/components/Charts/SparklineChart.tsx [hunk 1]
```diff
@@ -114,7 +114,7 @@ export class SparklineChart extends React.Component<Props, State> {
         domainPadding={{ y: 1 }}
       >
         <ChartAxis tickCount={15} style={hiddenAxisStyle} />
-        <ChartAxis dependentAxis={true} style={hiddenAxisStyle} />
+        <ChartAxis tickCount={2} dependentAxis={true} />
```
Comment (lucasponce) [inline src/components/Charts/SparklineChart.tsx:117]: This component is used also in the graph sidepanel, worth to check that we are not introduced side effects.

Comment (lucasponce) [reply src/components/Charts/SparklineChart.tsx:117]: I guess this is due this change:
image
(A minor empty y-axis)

Not a big deal, but I wonder if it could be better just to pass a property to differentiate one use or another.

Comment (leandroberetta) [reply src/components/Charts/SparklineChart.tsx:117]: Sure, sorry about that, I prefer to differentiate them, thanks @lucasponce

**Comment by lucasponce** [comment]
Hi @leandroberetta I could test it a little bit more and I may have a couple of comments, mostly style.

I suggest to put the unit in the y-axis:
image
image

"Ops" should be consistent with the metrics tab.

Another minor change that can be done in this PR, the title "traffic" is ambiguous and it was not changing for some time :) (sorry, I don't think I opened an issue), but this:
image

It could be better to change "Traffic" -> "Inbound traffic", it was not in the scope of this issue, but I think it'd be nice to update it.

Why ? In the overview is not collected the "Outbound traffic", i.e. in the "travel-control" namespace you should read "No inbound traffic" as "No traffic is a little bit ambiguos".

It's out of the scope of this issue if we want to bring the outbound traffic as well, that may be something to discuss in the future.

Another change, in the tooltips, I think it could be better to change "RPS" > "Total" in the blue one, and to change "Errors" > "4XX+5XX".

With these minor changes, I guess the original scope would be mostly done and it would improve the information that today is presented in the Overview page.

**Comment by leandroberetta** [comment]
@lucasponce yes, makes sense, thanks!.

**Comment by leandroberetta** [comment]
@lucasponce all the suggested changes were addressed.

overview_5

There is still something I'm figuring out how to solve, you can notice when the error series is re drew, is like for some period of time there weren't errors, i'm investigating the query, as it its made with promises and chunks.

**Comment by lucasponce** [comment]
> There is still something I'm figuring out how to solve, you can notice when the error series is re drew, is like for some period of time there weren't errors, i'm investigating the query, as it its made with promises and chunks.

I'll take a look, but I think it's the same query. I'll comment on the PR.

File: src/pages/Overview/OverviewPage.tsx [hunk 2]
```diff
@@ -84,7 +84,7 @@ const gridStyleList = style({
   marginTop: '0px'
 });
 
-const cardGridStyle = style({ borderTop: '2px solid #39a5dc', textAlign: 'center', marginTop: '0px' });
+const cardGridStyle = style({ borderTop: '2px solid #39a5dc', textAlign: 'center', marginTop: '10px' });
```
Comment (lucasponce) [inline src/pages/Overview/OverviewPage.tsx:87]: I'd recommend

```
const cardGridStyle = style({ borderTop: '2px solid #39a5dc', textAlign: 'center', marginTop: '0px', marginBottom: '10px' });
```

Then the style from the overview and the list pages would keep same top margin:
image
image

File: src/pages/Overview/OverviewCardSparkline.tsx [hunk 3]
```diff
@@ -5,35 +5,46 @@ import { Metric } from '../../types/Metrics';
...
   render() {
     if (this.props.metrics && this.props.metrics.length > 0) {
-      const data = toVCLines(this.props.metrics, 'rps', [PFColors.Blue400], 'time');
+      let series: VCLine<RichDataPoint>[] = [];
+
+      const data = toVCLine(this.props.metrics[0].datapoints, 'Total', PFColors.Blue400);
+      series.push(data);
+
+      if (this.props.errorMetrics && this.props.errorMetrics.length > 0) {
+        const dataErrors = toVCLine(this.props.errorMetrics[0].datapoints, '4XX+5XX', PFColors.Danger);
```
Comment (lucasponce) [inline src/pages/Overview/OverviewCardSparkline.tsx:28]: I'd suggest to '4XX+5XX' -> '4xx+5xx'.
I think typically uses lowercase for the "x"
image

**Review by lucasponce — CHANGES_REQUESTED** [review]
Ouch, I've found a problem: the Overview page can present the namespaces in three views (expanded, compact and list). It seems the table view has not updated the new error rates:

image

I think this should be also be fixed.

I see also a refresh in the error series in the default/expanded view, I'm also trying to see if there is a missing event in some of the charts components.

**Comment by lucasponce** [comment]
Found the refresh issue for the error series:
image

It needs to update the "errorMetrics" with the "previous.errorMetrics".

**Comment by lucasponce** [comment]
@leandroberetta this is the code that invokes the table in the overview page:

**Comment by leandroberetta** [comment]
> @leandroberetta this is the code that invokes the table in the overview page:
> image
> 
> I can try to send you a PR to your branch if you like.

@lucasponce thanks for the help, I've just committed the fix, now the error series appear in the List view:

list

I guess that with this last change, all suggestions are addressed.

**Comment by lucasponce** [comment]
@leandroberetta feature looks ok, I'm going to finish my review.
Would you mind to rebase with master ? Now that the axios bug is fixed I'd like to validate that the PR is working ok before to include QE on the review.

**Review by lucasponce — APPROVED** [review]
Feature and code looks ok.

**Comment by leandroberetta** [comment]
> @leandroberetta feature looks ok, I'm going to finish my review.
> Would you mind to rebase with master ? Now that the axios bug is fixed I'd like to validate that the PR is working ok before to include QE on the review.

@lucasponce it's rebased now. Thanks for the review.

**Comment by prachiyadav** [comment]
Test Case >> SWSQE-1987

**Review by prachiyadav — APPROVED** [review]
Screenshot from 2021-09-22 19-17-22
